Source/WebKit/UIProcess/API/Cocoa/_WKNotificationData.h

```diff
@@ -24,20 +24,53 @@
  */
 #import <WebKit/WKFoundation.h>
 
+@class WKSecurityOrigin;
+
 typedef NS_ENUM(NSUInteger, _WKNotificationAlert) {
     _WKNotificationAlertDefault,
     _WKNotificationAlertSilent,
     _WKNotificationAlertEnabled
 };
 
+typedef NS_ENUM(NSUInteger, _WKNotificationDirection) {
+    _WKNotificationDirectionAuto,
+    _WKNotificationDirectionLTR,
+    _WKNotificationDirectionRTL
+};
+
 WK_CLASS_AVAILABLE(macos(13.3), ios(16.4))
 @interface _WKNotificationData : NSObject
-@property (nonatomic, readonly, copy) NSString *title;
-@property (nonatomic, readonly, copy) NSString *body;
-@property (nonatomic, readonly, copy) NSString *origin;
-@property (nonatomic, readonly, copy) NSString *identifier;
-@property (nonatomic, readonly, copy) NSDictionary *userInfo;
+- (instancetype)init NS_UNAVAILABLE;
+
+@property (nonatomic, readonly) NSString *title;
+@property (nonatomic, readonly) _WKNotificationDirection dir;
+@property (nonatomic, readonly) NSString *lang;
+@property (nonatomic, readonly) NSString *body;
+@property (nonatomic, readonly) NSString *tag;
 @property (nonatomic, readonly) _WKNotificationAlert alert;
+@property (nonatomic, readonly) NSData *data;
+
+@property (nonatomic, readonly) NSString *origin;
+@property (nonatomic, readonly) NSURL *securityOrigin;
+@property (nonatomic, readonly) NSURL *serviceWorkerRegistrationURL;
+
+@property (nonatomic, readonly) NSString *identifier;
+@property (nonatomic, readonly) NSUUID *uuid;
+@property (nonatomic, readonly, copy) NSDictionary *userInfo;
+@end
+
+WK_CLASS_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA))
+@interface _WKMutableNotificationData : _WKNotificationData
```

Comment:
Can we just make `_WKNotificationData` properties readwrite? What's the benefit of `_WKMutableNotificationData`?

Reply:
Hmmmmm will stew on this offline.